design/design-doc-composition.md

```diff
@@ -0,0 +1,1400 @@
+  is deleted.
+
+Under the proposed design `spec.reclaimPolicy` exists at two levels:
+
+* At the managed resource level the reclaim policy determines what happens to
+  an external resource when its corresponding managed resource is deleted.
+* At the composite resource level the reclaim policy determines what happens to
+  the composite resource when it is bound to a resource requirement that is
+  deleted. There is no `spec.bindingPhase`, so a 'retained' composition becomes
+  available to subsequent resource requirements.
+
+This document recommends these fields be renamed once resource claims are
+removed in order to avoid any implication that they function identically to the
+reclaim policy of a persistent volume.
+
+### For Applications
+
+Application composition introduces new resource kinds (e.g. `Composition` and
+`ApplicationDefinition`) that do not affect existing applications built using
+the  templating-controller (aka "Template Stacks"). This document proposes that
+the templating-controller be marked as deprecated and its functionality merged
+into composition in a future release of Crossplane. This design may function
+alongside templated applications until that time.
+
+## Alternatives Considered
+
+_Many_ alternatives were considered before arriving at the design put forward by
+this document. While these alternatives were ultimately discarded, many of the
+concepts they propose survive in this proposal. Some alternatives were captured
+as design documents or sketches, including in rough historical order:
+
+* The [Aggregate Resources] design. This design addressed only infrastructure
+  composition, and attempted to build on rather than replace the class and claim
+  pattern. This design was abandoned when it became clear that it was untenable
+  for Crossplane to maintain a fixed set of portable, useful, resource claims.
+* The [ResourceClaimDefinition] sketch laid the foundation for the design put
+  forward by this document, but did not address application composition. It
+  continued to hold to the class and claim concept, while making it possible for
+  an infrastructure operator to define their own claims. It was abandoned in
+  part due to a desire to compose infrastructure at the cluster scope in order
+  to allow infrastructure administrators to define and reuse compositions that
+  they may not wish to allow application operators to invoke.
+* The [CompositeResourceDefinition] gist (and subsequent sketches in comments)
+  began to address the possibility of application composition, and of composing
+  infrastructure at the cluster scope. It can be considered a series of early
+  drafts that lead to this proposal.
+* The [define-your-claim] repository was the leading alternative to this
+  proposal. It maintained the concepts of resource classes, and offered stronger
+  typing and security in exchange for using many distinct resource class kinds
+  in the place of a single `Composition` kind.
+
+## Terminology
```

Comment:
Isn't _Terminology_ section used to be at the top?

Reply:
I've been both. In this design I prefer having it down here out of the way.